zerver/lib/actions.py

```diff
@@ -3000,24 +3000,22 @@ def make_dict(row):
 
 def do_process_attachment(message):
 
-    def extract_path_id(upload_string):
-        #We need to check if the string within the paranthesis contains a
-        #valid upload string
-        if re.search('[/\-]user[\-_]uploads[/\.-]', upload_string):
-            cleaned_path = re.sub('http://.*?/user_uploads/','', upload_string)
-            cleaned_path = re.sub('[\(\)]', '', cleaned_path)
+    def extract_path_id(url):
+        #This check is probably not needed anymore
+        if re.search('[/\-]user[\-_]uploads[/\.-]', url):
+            cleaned_path = re.sub('[/\-]user[\-_]uploads[/\.-]','', url)
+            #remove any extra '.' after file extension. These are probably added by the user
+            cleaned_path = re.sub('[.]+$','',cleaned_path, re.M)
             return cleaned_path
         else:
             return None
 
-    #get all pairs of paranthesis
-    p = re.compile('\(.*?\)')
-    possible_attachments = p.findall(message['message'].content)
+    atttachment_url_re = re.compile('[/\-]user[\-_]uploads[/\.-].*?(?=[ )]|\Z)')
```

Comment:
@timabbott Ive changed the regex to look for instances of /user_upload/. Ive added a test case to test it out too.